contracts/Booster.sol

```diff
@@ -594,20 +626,27 @@ contract Booster is ReentrancyGuardUpgradeable {
     }
 
     //claim fees from fee distro contract, put in lockers' reward contract
-    function earmarkFees() external returns (bool) {
+    function earmarkFees(address feeToken, bytes calldata _executionData) external returns (bool) {
+        // hash our execution data for comparison
+        bytes32 hashedExecutionData = keccak256(_executionData);
+        // enforce that the execution data is approved
+        require(hashedExecutionData == feeTokens[feeToken].executionHash, "!auth");
         //claim fee rewards
-        IStaker(staker).claimFees(feeDistro, feeToken);
-        //send fee rewards to reward contract
+        IStaker(staker).claimFees(feeTokens[feeToken].distro, feeToken, _executionData);
```

Comment:
as I understand _executionData will have the function signature without the inputs , so I think the inputs are missing here when pass it to function claimFee

Reply:
_executionData is both the function signature and the arguments.  This is how we enforce that the arguments passed into claimFees can't be tampered with <3